### ISSUE TITLE
Extremely slow page load

### ISSUE THREAD

**Issue by idagroup** [openprovider/Openprovider-WHMCS-domains#345]
https://github.com/openprovider/Openprovider-WHMCS-domains/blob/71684dce9fe6735bcb3e170fff591d81737d9fdf/modules/registrars/openprovider/hooks.php#L15C1-L15C36

Every page took at least 20 seconds to load, but I found the culprit and it is now back to ~2 second load times.

What I did:
In hooks.php, I moved `require_once(__DIR__ .'/init.php');` from line 15 to 7 and replaced it with `require_once '../../../init.php';`

**Comment by sapillai** [comment]
Hi @idagroup,

Thank you for sharing. 

I couldn't reproduce it in my test environment, but I will have our developers review this.

**Comment by idagroup** [comment]
Hey @sapillai, we're using the latest WHMCS on PHP 8.1 along with the ModulesGarden MultiBrand module, and that might be the reason for those delays. I don't think we need to use __DIR__ since the path between the two files stays consistent.

**Comment by DennisSkov** [comment]
@idagroup replacing that line loads the init.php from the WHMCS root instead of the init.php in the module folder. That is most certainly going to cause issues.

**Comment by idagroup** [comment]
@DennisSkov, you are correct that it appears to be loading the incorrect init.php file. However, this change has resolved the issue of slow page loads, and so far, I haven't encountered any functionality issues, even with the hooks.php file completely disabled. 

Do you know what role hooks.php plays and what potential effects its absence might have?

**Comment by DennisSkov** [comment]
In WHMCS, the `hooks.php` file is used for hooks (hookable events). It's a way to execute code at specific events, fx when a domain is successfully transferred.

**Comment by idagroup** [comment]
Thanks, @DennisSkov. We often use hooks, but I wasn't quite sure what this particular one is doing. Additionally, WHMCS already offers a feature to monitor pending registration transfers and automatically update the status to "Active" once the domain is with the chosen registrar. Given this, I believe it should be safe to remove hooks.php from the openprovider folder to prevent it from interfering and slowing down the client area experience.

**Comment by DennisSkov** [comment]
It's not entirely safe, but your use case might not depend on these hooks.
The hooks are not defined in `hooks.php` - it's just the file that initiates them. They're defined in their relevant controllers here: https://github.com/openprovider/Openprovider-WHMCS-domains/tree/master/modules/registrars/openprovider/Controllers/Hooks.

The module uses hooks to output required stuff for certain TLDs (es and pt) in the cart.
It also hides the registrar lock option for TLDs that does not support it.

**Comment by idagroup** [comment]
Oh, interesting! I'll take a look at the controller files when I have a free moment. As for the issue with hiding registrar locks, I've addressed it with the template change discussed [here](https://github.com/openprovider/Openprovider-WHMCS-domains?tab=readme-ov-file#configure-tlds-which-dont-allow-locking-nl-be-eu-es)

**Comment by sapillai** [comment]
Hi @idagroup,

I am closing this issue as the modification loads a different init.php file as pointed out by @DennisSkov and using magic constant `__DIR__` is the recommended option for stability, readability and maintainability. Using paths like ../../../ can be confusing and error-prone as the project structure evolves or if the the directory structure changes.

For slow page loads, I suggest to follow steps in [Optimize Performance](https://docs.whmcs.com/system/system-tutorials/optimize-performance/) that can help you maintain faster speeds.
